lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -542,7 +542,12 @@ function dosomething_campaign_get_campaigns_doing($uid = NULL) {
 function dosomething_campaign_get_campaign_block_vars($nid, $image_size = '740x480', $source = NULL) {
   $node = node_load($nid);
   $clc = dosomething_helpers_get_current_language_content_code();
-  $path_alias = drupal_get_path_alias('node/' . $nid);
+
+  global $user;
+  $language = dosomething_global_get_language($user, $node);
```

Comment:
shouldn't this be the node lang? what if the campaign doesn't exist in the user's lang?